Src/BootCamp.Chapter/Factory.cs

```diff
@@ -5,40 +5,34 @@ namespace BootCamp.Chapter
 {
     public abstract class Factory
     {
-        public abstract DesktopComputer Assemble();
-        public Body AddBody(string brand)
+        public abstract string Brand();
+        public DesktopComputer Assemble()
         {
-            PrintAddMsg("Body", brand);
-            return new Body(brand + "Body");
-        }
-        public Cpu AddCpu(string brand)
-        {
-            PrintAddMsg("Cpu", brand);
-            return new Cpu(brand + "Cpu");
-        }
-        public Gpu AddGpu(string brand)
-        {
-            PrintAddMsg("Gpu", brand);
-            return new Gpu(brand + "Gpu");
-        }
-        public HardDisk AddHardDisk(string brand)
-        {
-            PrintAddMsg("HardDisk", brand);
-            return new HardDisk(brand + "HardDisk");
-        }
-        public Motherboard AddMotherboard(string brand)
-        {
-            PrintAddMsg("Motherboard", brand);
-            return new Motherboard(brand + "Motherboard");
-        }
-        public Ram AddRam(string brand)
-        {
-            PrintAddMsg("Ram", brand);
-            return new Ram(brand + "Ram");
-        }
-        private void PrintAddMsg(string component, string brand)
-        {
-            Console.WriteLine($"Adding {component} to {brand}PC.");
+            DesktopComputer PC = new DesktopComputer();
+            PC.Body = InstallBody();
```

Comment:
Does InstallBody installs a body? No- it just creates it. You should pass a computer and then install the new piece to the passed computer. So please redo based on that.

Reply:
Oké i think i understand it now. please review again. small change.